### PR TITLE
Delete my profile implemented

### LINKED ISSUE #11: Implement delete my profile endpoint
**Description:**
This issue involves implementing an API to update the profile details. The endpoint should allow clients to make a DELETE request to `{{baseUrl}}/api/me/profile/{{profileId}}`.

**Tasks:**
1. Create a controller for the `/controllers/profile` endpoint in the backend (create the route profile if not created).
2. Parse and validate the request body to ensure it matches the expected format.
3. Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
5. Write unit tests to validate the functionality and correctness of the endpoint.

API documentation: https://documenter.getpostman.com/view/27421496/2s93m1a4ac#8744a3ee-970f-489a-853d-8b23fdee8de3

ER diagram: https://drive.google.com/file/d/11KMgdNu2mSAm0Ner8UsSPQpZJS8QNqYc/view 

**Acceptance Criteria:**
- The profile API endpoint `/controllers/profile` is implemented via a DELETE request.
- The request body is properly parsed and validated for the required format.
- Appropriate error handling is implemented, providing meaningful error messages and correct response status codes.
- Unit tests are written to validate the functionality and correctness of the endpoint.

**Additional Information:**
No

**Related Dependencies or References:**
No

### PR DESCRIPTION
**Pull request by Danishkar**
## Purpose

The purpose of this PR is to fix #11<issue-number>

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [ ] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/controllers/profile.controller.ts [hunk 1]
```diff
@@ -48,3 +48,22 @@ export const updateProfileHandler = async (
...
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const userId = req.params.uuid
+    const user = await getProfile(req)
+    if (!user || user.uuid !== userId) {
+      res.status(404).json({ message: 'Profile not found' })
+    } else {
+      await deleteProfile(user)
```
Comment (anjula-sack) [inline src/controllers/profile.controller.ts:62]:
```suggestion
      await deleteProfile(user.uuid)
```

File: src/services/profile.service.ts [hunk 2]
```diff
@@ -39,3 +39,14 @@ export const updateProfile = async (
 
   return savedProfile
 }
+
+export const deleteProfile = async (user: Profile): Promise<void> => {
```
Comment (anjula-sack) [inline src/services/profile.service.ts:43]:
```suggestion
export const deleteProfile = async (userId: string): Promise<void> => {
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]
can you write tests as well? @Danishkar

File: src/controllers/profile.controller.ts [hunk 3]
```diff
@@ -48,3 +48,22 @@ export const updateProfileHandler = async (
...
   }
 }
+
+export const deleteProfileHandler = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const userId = req.params.uuid
+    const user = await getProfile(req)
```
Comment (anjula-sack) [inline src/controllers/profile.controller.ts:58]:
```suggestion
    const user = req.user
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by anjula-sack** [comment]
Hi @Danishkar, can you fix the conflicts?

**Comment by Danishkar** [comment]
> Hi @Danishkar, can you fix the conflicts?

yea sure

**Comment by Danishkar** [comment]
> can you write tests as well? @Danishkar

i will work on that

**Comment by anjula-sack** [comment]
> > Hi @Danishkar, can you fix the conflicts?
> 
> yea sure

Can you fix the lint errors as well? Run `npm run lint:fix`

**Comment by Danishkar** [comment]
> npm run lint:fix

done

**Review by anjula-sack — CHANGES_REQUESTED** [review]
@Danishkar can you run lint fix again? The CI is failing.

**Comment by anjula-sack** [comment]
@Danishkar the tests are failing

**Comment by Danishkar** [comment]
> @Danishkar the tests are failing

yea bro I am new to this jest testing I will check on that.

File: src/controllers/profile.controller.ts [hunk 4]
```diff
@@ -46,3 +45,22 @@ export const updateProfileHandler = async (
     }
   }
 }
+
+export const deleteProfileHandler = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const userId = req.params.uuid
```
Comment (anjula-sack) [inline src/controllers/profile.controller.ts:54]:
```suggestion
```

File: src/controllers/profile.controller.ts [hunk 5]
```diff
@@ -46,3 +45,22 @@ export const updateProfileHandler = async (
...
+    const user = req.user as Profile
+    if (!user || user.uuid !== userId) {
```
Comment (anjula-sack) [inline src/controllers/profile.controller.ts:56]:
```suggestion
    if (!user) {
```

File: src/routes/profile/profile.route.ts [hunk 6]
```diff
@@ -9,5 +10,6 @@ const profileRouter = express.Router()
 
 profileRouter.get('/profile', requireAuth, getProfileHandler)
 profileRouter.put('/profile', requireAuth, updateProfileHandler)
+profileRouter.delete('/profile/:uuid', requireAuth, deleteProfileHandler)
```
Comment (anjula-sack) [inline src/routes/profile/profile.route.ts:13]:
```suggestion
profileRouter.delete('/profile', requireAuth, deleteProfileHandler)
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]
Requested some changes

File: src/routes/profile/profile.route.test.ts [hunk 7]
```diff
@@ -57,10 +57,18 @@ describe('profile', () => {
 
       await agent.put('/api/me/profile').send(updatedProfile).expect(200)
     })
-
     it('should return a 401 when a valid access token is not provided', async () => {
       await supertest(server).put('/api/me/profile').send({}).expect(401)
     })
+  })
```
Comment (anjula-sack) [inline src/routes/profile/profile.route.test.ts:63]:
```suggestion
  })
  
```

File: src/routes/profile/profile.route.test.ts [hunk 8]
```diff
@@ -57,10 +57,18 @@ describe('profile', () => {
 
       await agent.put('/api/me/profile').send(updatedProfile).expect(200)
     })
```
Comment (anjula-sack) [inline src/routes/profile/profile.route.test.ts:59]:
```suggestion
    })
    
```

File: src/routes/profile/profile.route.test.ts [hunk 9]
```diff
@@ -57,10 +57,20 @@ describe('profile', () => {
...
-
+    
     it('should return a 401 when a valid access token is not provided', async () => {
       await supertest(server).put('/api/me/profile').send({}).expect(401)
     })
+  })
+
```
Comment (anjula-sack) [inline src/routes/profile/profile.route.test.ts:65]:
```suggestion
 
```

File: src/routes/profile/profile.route.test.ts [hunk 10]
```diff
@@ -57,10 +57,20 @@ describe('profile', () => {
 
       await agent.put('/api/me/profile').send(updatedProfile).expect(200)
     })
-
+
```
Comment (anjula-sack) [inline src/routes/profile/profile.route.test.ts:60]:
```suggestion
 
```

**Comment by anjula-sack** [comment]
@Danishkar can you take a pull and fix the lint issues @Danishkar

**Review by anjula-sack — APPROVED** [review]
Good job! @Danishkar

**Pull request merged**